core/src/main/java/org/frankframework/senders/RuntimeSenderWrapper.java

```diff
@@ -0,0 +1,85 @@
+/*
+   Copyright 2024 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package org.frankframework.senders;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.frankframework.configuration.ConfigurationException;
+import org.frankframework.core.ISender;
+import org.frankframework.core.ISenderWithParameters;
+import org.frankframework.core.PipeLineSession;
+import org.frankframework.core.SenderException;
+import org.frankframework.core.SenderResult;
+import org.frankframework.core.TimeoutException;
+import org.frankframework.statistics.HasStatistics;
+import org.frankframework.statistics.StatisticsKeeperIterationHandler;
+import org.frankframework.stream.Message;
+
+/**
+ * Wrapper for senders, that opens 'runtime' before each sender action, and closes afterwards. Prevents long open connections inside Senders and possible connection failures. 
+ *
+ * @ff.parameters any parameters defined on the SenderWrapper will be handed to the sender, if this is a {@link ISenderWithParameters}
+ *
+ * @author  Niels Meijer
+ */
+public class RuntimeSenderWrapper extends SenderWrapperBase {
```

Comment:
I don't like the name because it doesn't tell you what it does, but I don't have a better name yet.

- `OpenOnDemandSenderWrapper`?
- `ConnectOnDemandSenderWrapper`?
- `RuntimeOpenSenderWrapper`?
- `DynamicOpenSenderWrapper`?

Reply:
Yes, I agree with you. I like this one the best `ConnectOnDemandSenderWrapper` or what about `OpenCloseSenderWrapper`? That's exactly what happens. Perhaps let Sena validate it, since she is exactly a person who directly needs to understand it.